srtcore/core.cpp

```diff
@@ -2583,6 +2597,19 @@ bool CUDT::interpretSrtHandshake(const CHandShake& hs, const CPacket& hspkt, uin
 
             break;
         }
+#else
+        // When encryption is not enabled at compile time, behave as if encryption wasn't set,
+        // so accordingly to StrictEncryption flag.
+
+        if (m_bOPT_StrictEncryption)
+        {
+            LOGC(mglog.Error, log << "HS KMREQ: Peer declares encryption, but agent didn't enable it at compile time - rejecting per strict requirement");
+            return false;
+        }
+
+        LOGC(mglog.Error, log << "HS KMREQ: Peer declares encryption, but agent didn't enable it at compile time - still allowing connection.");
+        encrypted = true;
+#endif
```

Comment:
So in case strict encryption mode is turned on, the connection should be rejected. While with strict encryption turned off a connection should be established, but will have a status BAD_SECRET or UNPROTECTED.

Reply:
Should behave the same way as in case when the password wasn't set, so rather not BADSECRET.